templates/.github/workflows/automerge.yml

```diff
@@ -4,9 +4,32 @@
 on: [ pull_request ]
 
 jobs:
-  automerge:
+  automerge-check:
     if: github.event.pull_request.user.login == 'web3-bot'
     runs-on: ubuntu-latest
+    outputs:
+      status: ${{ steps.should-automerge.outputs.status }}
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+    - name: Check if we should automerge
+      id: should-automerge
+      run: |
+        for commit in $(git rev-list --first-parent origin/${{ github.event.pull_request.base.ref }}..${{ github.event.pull_request.head.sha }}); do
+          committer=$(git show --format=$'%ce' -s $commit)
+          echo "Committer: $committer"
+          if [[ "$committer" != "web3-bot@users.noreply.github.com" ]]; then
```

Comment:
just so we're on the same page, this is a convenience feature (so we don't accidentally merge things before we're ready) not a security feature.

Reply:
Yes. Security is dependent on the `if: github.event.pull_request.user.login == 'web3-bot'`.

While git commit author / committer can trivially be faked by an attacker, GitHub won't allow a user to open a PR for another user.